lib/dashboard/SingleWidgetDashboardSegment.ts

```diff
@@ -1,30 +1,37 @@
 import { IWidget } from "aws-cdk-lib/aws-cloudwatch";
 
 import { IDashboardSegment } from "./DashboardSegment";
+import { DefaultDashboards } from "./DefaultDashboardFactory";
+import { IDynamicDashboardSegment } from "./DynamicDashboardSegment";
 
-export class SingleWidgetDashboardSegment implements IDashboardSegment {
+export class SingleWidgetDashboardSegment
+  implements IDashboardSegment, IDynamicDashboardSegment
+{
   protected readonly widget: IWidget;
-  protected readonly addToSummary: boolean;
-  protected readonly addToAlarm: boolean;
 
-  constructor(widget: IWidget, addToSummary?: boolean, addToAlarm?: boolean) {
+  constructor(widget: IWidget) {
```

Comment:
Should this accept a list of dashboard names that we'd want the widget to appear on? As-is, this restricts `SingleWidgetDashboardSegment` to only be useful for the default dashboards.